hoomd/mpcd/SphereGeometry.h

```diff
@@ -0,0 +1,209 @@
+
+#ifndef MPCD_SPHERE_GEOMETRY_H_
+#define MPCD_SPHERE_GEOMETRY_H_
+
+#include "BoundaryCondition.h"
+
+#include "hoomd/HOOMDMath.h"
+#include "hoomd/BoxDim.h"
+
+#include <cmath>
+
+#ifdef NVCC
+#define HOSTDEVICE __host__ __device__ inline
+#else
+#define HOSTDEVICE inline __attribute__((always_inline))
+#include <string>
+#endif // NVCC
+
+namespace mpcd
+{
+namespace detail
+{
+//! Spherical droplet geometry
+/*!
+ * TODO: write proper description of what is being done in the class below
+ */
+class __attribute__((visibility("default"))) SphereGeometry
+    {
+    public:
+        //! Constructor
+        /*!
+         * \param R confinement radius
+         * \param bc Boundary condition at the wall (slip or no-slip)
+         */
+        HOSTDEVICE SphereGeometry(Scalar R, boundary bc)
+            : m_R(R), m_bc(bc)
```

Comment:
You use `R*R` enough times that you may also want to store it as a member variable:
```suggestion
            : m_R(R), m_R2(R*R), m_bc(bc)
```
and use it everywhere you use R*R. This is a more important consideration on CPUs than GPUs.